pravega-client-examples/src/main/java/io/pravega/example/streamprocessing/NonRecoverableSingleThreadedProcessor.java

```diff
@@ -0,0 +1,142 @@
+        final ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                .stream(Stream.of(scope, inputStreamName))
+                .build();
+        try (ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, controllerURI)) {
+            readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
+        }
+
+        try (ClientFactory clientFactory = ClientFactory.withScope(scope, controllerURI);
+             EventStreamReader<String> reader = clientFactory.createReader(
+                     "reader",
+                     readerGroup,
+                     new UTF8StringSerializer(),
+                     ReaderConfig.builder().build());
+             EventStreamWriter<String> writer = clientFactory.createEventWriter(
+                     outputStreamName,
+                     new UTF8StringSerializer(),
+                     EventWriterConfig.builder().build())) {
+
+            long eventCounter = 0;
+
+            for (; ; ) {
+                // Read input event.
+                EventRead<String> eventRead = reader.readNextEvent(READER_TIMEOUT_MS);
+                log.debug("readEvents: eventRead={}", eventRead);
+
+                if (eventRead.getEvent() != null) {
+                    eventCounter++;
+                    log.debug("Read eventCounter={}, event={}", String.format("%06d", eventCounter), eventRead.getEvent());
+
+                    // Parse input event.
+                    String[] cols = eventRead.getEvent().split(",");
+                    long generatedEventCounter = Long.parseLong(cols[0]);
+                    String routingKey = cols[1];
+                    long intData = Long.parseLong(cols[2]);
+                    long generatedSum = Long.parseLong(cols[3]);
+                    String generatedTimestampStr = cols[4];
+
+                    // Process the input event.
+                    String processedTimestampStr = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX").format(new Date());
+
+                    // Build the output event.
+                    String message = String.join(",",
+                            String.format("%06d", generatedEventCounter),
+                            String.format("%06d", eventCounter),
+                            routingKey,
+                            String.format("%02d", intData),
+                            String.format("%08d", generatedSum),
+                            String.format("%03d", 0),
+                            generatedTimestampStr,
+                            processedTimestampStr,
+                            "");
+
+                    // Write the output event.
+                    log.info("eventCounter={}, event={}",
+                            String.format("%06d", eventCounter),
+                            message);
+                    final CompletableFuture writeFuture = writer.writeEvent(routingKey, message);
```

Comment:
We don't need to create a variable here to capture the result?

Reply:
I wanted to make it clear to the reader that writeEvent returns a future and returning doesn't indicate that the event was actually written. I added a comment to the code to clarify this.